### PR TITLE
Enable CoreDNS's autopath feature

### PR DESCRIPTION
**Pull request by linki**
Enable CoreDNS's [autopath](https://coredns.io/plugins/autopath/) feature for Kubernetes.

So far I didn't find any good indication whether one should also change the `resolv.conf`.

With this setup a lookup to an external name like `google.com` would go first to `google.com.kube-system.svc.cluster.local` which `autopath`-enabled CoreDNS is able to answer with a CNAME to `google.com` instead of an NXDOMAIN, thus not needing the client to do any further requests for the other search domains.

Using `pods verified` is required when using `autopath`.

Interesting read about Kubernetes' DNS issues: https://blog.quentin-machu.fr/2018/06/24/5-15s-dns-lookups-on-kubernetes/

### CODE REVIEW

**Comment by szuecs** [comment]
This should be done for coredns as daemonset, not?

**Comment by njuettner** [comment]
Probably we have to adjust the memory.
`This option requires substantially more memory than in insecure mode, since it will maintain a watch on all pods.`

**Comment by njuettner** [comment]
What about having a cluster default config item for `coredns_cpu` and `coredns_mem`?

And change the current deployment to a daemonset like @szuecs suggested. Additionally to that, fixing the QoS from `Burstable` to  `Guaranteed` would make sense to me (`CPU`limit is missing).

/cc @linki @szuecs

**Comment by linki** [comment]
> This should be done for coredns as daemonset, not?

That doesn't matter. It applies to both setups.

**Comment by linki** [comment]
@njuettner Makes sense to me. The DaemonSet/Deployment change is in [another PR](https://github.com/zalando-incubator/kubernetes-on-aws/pull/1573) as it's not directly related. Although, we can think about not using `autopath` with the DaemonSet setup, but I don't see a reason to if autopath works well.

**Comment by njuettner** [comment]
👍

**Comment by jrake-revelant** [comment]
👍

File: cluster/manifests/coredns/configmap.yaml [hunk 1]
```diff
@@ -11,10 +11,18 @@ data:
         errors
         health
         kubernetes cluster.local in-addr.arpa ip6.arpa {
+            resyncperiod 1m
```
Comment (linki) [inline cluster/manifests/coredns/configmap.yaml]: @njuettner Should we do this in a separate PR or move into the `if`? This makes this PR essentially contain two changes (1 guarded by a config item and this one). The default sync period is `5m`, right?

Comment (njuettner) [reply cluster/manifests/coredns/configmap.yaml]: Correct, the default would be 5 minutes. I'll add an `if`.

Comment (linki) [reply cluster/manifests/coredns/configmap.yaml]: Ok. I'm always unsure about this stuff. Just want to mention that's also possible to use a config item and set its default value to `5m` to avoid the `if` or to leave it out entirely for now.

**Comment by mikkeloscar** [comment]
I think for this feature it would be very helpful to do a load test which shows the impact, also in terms of extra resources required on the coredns side.

e.g. similar to what we did for skipper as a deployment: #1457

**Comment by szuecs** [comment]
Please see @mikkeloscar comment, I set this to do-not-merge

**Comment by linki** [comment]
Please note that this feature is guarded by a config item and wouldn't change anything when rolled out.

**Comment by linki** [comment]
We're trying CoreDNS as DaemonSet first. Consider enabling `autopath` if it's not enough.
